client/lib/url/omit-url-params.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Internal dependencies
+ */
+import { URL as TypedURL } from 'types';
+import { Falsey } from 'utility-types';
+
+/**
+ * Removes given params from a url.
+ *
+ * @param  url URL to be cleaned
+ * @param  paramsToOmit The collection of params or single param to reject
+ * @return Url less the omitted params.
+ */
+export default function omitUrlParams( url: Falsey, paramsToOmit: string | string[] ): null;
+export default function omitUrlParams( url: TypedURL, paramsToOmit: string | string[] ): TypedURL;
+export default function omitUrlParams(
+	url: TypedURL | Falsey,
+	paramsToOmit: string | string[]
+): TypedURL | null {
+	if ( ! url ) {
+		return null;
+	}
+
+	let toOmit: string[];
+
+	if ( typeof paramsToOmit === 'string' ) {
+		toOmit = [ paramsToOmit as string ];
+	} else {
+		toOmit = paramsToOmit;
+	}
+
+	const parsed = new URL( url );
```

Comment:
What happens with malformed or relative URLs? I'd expect this to throw. Is that the case and is it a change in behavior? Will you add some tests around that?

Reply:
Thank you for pointing that out! It seems that node's `parse` is extremely lenient here, and will happily handle things that aren't URLs, as long as there's a `?foo=bar` in there. Ah, how fun.

In order to try to get compatibility as high as possible, I'll need to place this PR on hold while I work on some new additions to `lib/url`.

Reply:
The current code should handle any valid URL, whether absolute or having any level of relative-ness. The tests have been expanded to ensure that.

Note that the PR will fail for now, as it's dependent on #37068.